plugins/vbe/fu-vbe-simple-device.c

```diff
@@ -0,0 +1,901 @@
+ * 512 bytes of the image (which must be present in the image) are skipped and
+ * the bytes after that are written to the store offset.
+ * @fd: File descriptor, if the device is open
+ * @vbe_fname: Filename of the VBE state file
+ * @state: State of this update method
+ */
+struct _FuVbeSimpleDevice {
+	FuVbeDevice parent_instance;
+	const gchar *storage;
+	gchar *devname;
+	off_t area_start;
+	off_t area_size;
+	gint skip_offset;
+	gint fd;
+	gchar *vbe_fname;
+	struct vbe_simple_state state;
+};
+
+G_DEFINE_TYPE(FuVbeSimpleDevice, fu_vbe_simple_device, FU_TYPE_VBE_DEVICE)
+
+/**
+ * trailing_strtoln_end() - Get the number suffix at the end of a string
+ *
+ * Decodes a string like "abc123" into its two parts so that the base string
+ * "abc" and the trailing number "123" can be obtained.
+ *
+ * For this example, with end == NULL, the function returns 123 and *endp
+ * returns a pointer to the '1' character.
+ *
+ * @str: String to parse
+ * @end: End of string to parse, or NULL to parse the whole string
+ * @endp: Returns a pointer to the first character of the numeric suffix. If
+ *	no number is found, this is @end if not NULL, else its points to the
+ *	string's nul terminator
+ * Returns: the numeric value of the suffix, or -1 if none found
+ */
+static gint
+trailing_strtoln_end(const gchar *str, const gchar *end, gchar const **endp)
+{
+	const gchar *p;
+
+	if (!end)
+		end = str + strlen(str);
+	p = end - 1;
+	if (p > str && isdigit(*p)) {
+		do {
+			if (!isdigit(p[-1])) {
+				if (endp)
+					*endp = p;
+				return atoi(p);
+			}
+		} while (--p > str);
+	}
+	if (endp)
+		*endp = end;
+
+	return -1;
+}
+
+/**
+ * fdt_get_u32() - Get a 32-bit integer value from the device tree
+ *
+ * @fdt: Device tree to read from
+ * @node: Node offset to read from
+ * @prop_name: Name of property to read
+ * Returns: integer value, if found and of the correct size, else -1
+ */
+static long
+fdt_get_u32(const gchar *fdt, gint node, const gchar *prop_name)
+{
+	const fdt32_t *val;
+	gint len;
+
+	val = fdt_getprop(fdt, node, prop_name, &len);
+	if (!val || len != sizeof(fdt32_t))
+		return -1;
+
+	return fdt32_to_cpu(*val);
+}
+
+/**
+ * fdt_get_u64() - Get a 64-bit integer value from the device tree
+ *
+ * @fdt: Device tree to read from
+ * @node: Node offset to read from
+ * @prop_name: Name of property to read
+ * Returns: integer value, if found and of the correct size, else -1
+ */
+static long
+fdt_get_u64(const gchar *fdt, gint node, const gchar *prop_name)
+{
+	const fdt64_t *val;
+	gint len;
+
+	val = fdt_getprop(fdt, node, prop_name, &len);
+	if (!val || len != sizeof(fdt64_t))
+		return -1;
+
+	return fdt64_to_cpu(*val);
+}
+
+static gboolean
+fu_vbe_simple_device_probe(FuDevice *self, GError **error)
+{
+	struct _FuVbeSimpleDevice *dev = FU_VBE_SIMPLE_DEVICE(self);
+	FuVbeDevice *vdev;
+	const void *fdt;
+	gint devnum, len;
+	gint node;
+
+	vdev = FU_VBE_DEVICE(self);
+	g_return_val_if_fail(FU_IS_VBE_DEVICE(self), FALSE);
```

Comment:
I think this needs to come before setting `vdev` in the line above

Reply:
done

Reply:
> I notice fedora CI fails, did you lose the changes to `contrib/ci/dependencies.xml` in a rebase?

Yes sadly I did. I just noticed it bringing up a new machine. Done